src/solc_0.5/Land/erc721/LandBaseTokenV3.sol

```diff
@@ -0,0 +1,553 @@
+            uint256 counter = 0;
+            for (uint256 j = 0; j < sizes.length; j++) {
+                uint256 size = sizes[j];
+                for (uint256 i = 0; i < size*size; i++) {
+                    ids[counter] = _idInPath(i, size, xs[j], ys[j]);
+                    counter++;
+                }
+            }
+            require(
+                _checkOnERC721BatchReceived(metaTx ? from : msg.sender, from, to, ids, data),
+                "erc721 batch transfer rejected by to"
+            );
+        }
+    }
+
+    function _transferQuad(address from, address to, uint256 size, uint256 x, uint256 y) internal {
+        if (size == 1) {
+            uint256 id1x1 = x + y * GRID_SIZE;
+            address owner = _ownerOf(id1x1);
+            require(owner != address(0), "token does not exist");
+            require(owner == from, "not owner in _transferQuad");
+            _owners[id1x1] = uint256(to);
+        } else {
+            _regroup(from, to, size, x, y);
+        }
+        for (uint256 i = 0; i < size*size; i++) {
+            emit Transfer(from, to, _idInPath(i, size, x, y));
+        }
+    }
+
+    function _checkAndClear(address from, uint256 id) internal returns(bool) {
+        uint256 owner = _owners[id];
+        if (owner != 0) {
+            require(address(owner) == from, "not owner");
+            _owners[id] = 0;
+            return true;
+        }
+        return false;
+    }
+
+    function _regroup(address from, address to, uint256 size, uint256 x, uint256 y) internal {
+        require(x % size == 0 && y % size == 0, "Invalid coordinates");
+        require(x <= GRID_SIZE - size && y <= GRID_SIZE - size, "Out of bounds");
+
+        if (size == 3) {
+            _regroup3x3(from, to, x, y, true);
+        } else if (size == 6) {
+            _regroup6x6(from, to, x, y, true);
+        } else if (size == 12) {
+            _regroup12x12(from, to, x, y, true);
+        } else if (size == 24) {
+            _regroup24x24(from, to, x, y, true);
+        } else {
+            require(false, "Invalid size");
+        }
+    }
+
+    function exists(uint256 _id) public view returns(bool){
+        uint256 quadType = uint8(_id >> 248);
+        if(quadType == 0){
+            return _exists1x1(_id);
+        }else if(quadType == 1){
+            require(getX(_id)%3 == 0 && getY(_id)%3 == 0, "LandBaseTokenV2: Invalid Id");
+            return _exists3x3(_id);
+        }else if(quadType == 2){
+            require(getX(_id)%6 == 0 && getY(_id)%6 == 0, "LandBaseTokenV2: Invalid Id");
+            return _exists6x6(_id);
+        }else if(quadType == 3){
+            require(getX(_id)%12 == 0 && getY(_id)%12 == 0, "LandBaseTokenV2: Invalid Id");
+            return _exists12x12(_id);
+        }else if(quadType == 4){
+            require(getX(_id)%24 == 0 && getY(_id)%24 == 0, "LandBaseTokenV2: Invalid Id");
+            return _exists24x24(_id);
+        }
```

Comment:
Can we use the same idea than in `_ownerOfQuad` to handle all the size and avoi all the exists*x* functions ?